xbmc/addons/kodi-dev-kit/include/kodi/addon-instance/Game.h

```diff
@@ -1179,6 +1307,98 @@ class ATTR_DLL_LOCAL CInstanceGame : public IAddonInstance
         ->SetCheat(index, enabled, code);
   }
 
+  inline static GAME_ERROR ADDON_RCGenerateHashFromFile(const AddonInstance_Game* instance,
+                                                        char** hash,
+                                                        unsigned int consoleID,
+                                                        const char* filePath)
+  {
+    std::string cppHash;
+
+    GAME_ERROR ret = static_cast<CInstanceGame*>(instance->toAddon->addonInstance)
+                         ->RCGenerateHashFromFile(cppHash, consoleID, filePath);
+    if (!cppHash.empty() && hash)
+    {
+      *hash = strdup(cppHash.c_str());
+    }
+    return ret;
+  }
+
+  inline static GAME_ERROR ADDON_RCGetGameIDUrl(const AddonInstance_Game* instance,
+                                                char** url,
+                                                const char* hash)
+  {
+    std::string cppUrl;
+    GAME_ERROR ret =
+        static_cast<CInstanceGame*>(instance->toAddon->addonInstance)->RCGetGameIDUrl(cppUrl, hash);
+    if (!cppUrl.empty() && url)
+    {
+      *url = strdup(cppUrl.c_str());
+    }
+    return ret;
+  }
+
+  inline static GAME_ERROR ADDON_RCGetPatchFileUrl(const AddonInstance_Game* instance,
+                                                   char** url,
+                                                   const char* username,
+                                                   const char* token,
+                                                   unsigned gameID)
```

Comment:
Also then here and on Kodi side to `unsigned int`, not only on class function `virtual GAME_ERROR RCGetPatchFileUrl`.

Reply:
Oops missed two unsigned params, fixed

Reply:
No problem :smile: 

Can you then do a rebase and to to resolve the conflicts and after fix about @lrusak comments, to cleanup the commit history. With the `fixup!` is tempoary OK, only not good as final :sweat_smile:.